lib/note-list/decorators.tsx

```diff
@@ -1,27 +1,96 @@
 import React from 'react';
 import { escapeRegExp } from 'lodash';
-import replaceToArray from 'string-replace-to-array';
+import removeAccents from '../utils/remove-accents';
 
 import { withoutTags } from '../utils/filter-notes';
 
-export const decorateWith = (decorators, text) =>
-  decorators.length > 0
-    ? decorators
-        .reduce((output, { filter, replacer }) => {
-          const searchText = 'string' === typeof filter && withoutTags(filter);
-          const pattern =
-            searchText && searchText.length > 0
-              ? new RegExp(escapeRegExp(searchText), 'gi')
-              : filter;
-
-          return replaceToArray(output, pattern, replacer);
-        }, text)
-        .map((chunk, key) =>
-          chunk && 'string' !== typeof chunk
-            ? React.cloneElement(chunk, { key })
-            : chunk
-        )
-    : text;
+function replaceToArray(string, regexp, newValueOrFn) {
+  console.log(string);
+  var output = [];
+
+  var replacerIsFn = typeof newValueOrFn === 'function';
+
+  var storedLastIndex = regexp.lastIndex;
+  regexp.lastIndex = 0;
+
+  var result;
+  var lastIndex = 0;
+  while ((result = regexp.exec(string))) {
+    var index = result.index;
+
+    if (result[0] === '') {
+      // When the regexp is an empty string
+      // we still want to advance our cursor to the next item.
+      // This is the behavior of String.replace.
+      regexp.lastIndex++;
+    }
+
+    if (index !== lastIndex) {
+      output.push(string.substring(lastIndex, index));
```

Comment:
I believe that this is going to produce some odd results. the act of removing parts of the text will shift the array indices of the search results. if there are lots of accents in the text I think this could get pretty far away from where it should be.

this is hard 😄